### ISSUE TITLE
Document conversion process

### ISSUE THREAD

**Issue by kadel** [rtnpro/henge#8]
It would be nice to have document describing how docker-compose directive are mapped to Kubernetes and OpenShift objects.
This document can also show how conversion to Kubernetes differs from conversion to OpenShift.

**Comment by pradeepto** [comment]
+1 to this idea. Very important. Thanks @kadel.

**Comment by rtnpro** [comment]
This tool should have been light, standalone. But, because of the dependency on ``libcompose``, ``openshift``, ``kubernetes`` repos, it has indirectly become a bulkier tool, and this fact shows up in the increased build time of this application, with so much less code on our side.

All we are doing is taking one file and producing another file from it, kind of a mapping (one-to-one or complicated), but it's at the end, a mapping. Because of the huge dependency chain of this library/tool, it will be difficult to promote it as a library for consumption in Openshift.

Let me know, what your thoughts are.

**Comment by kadel** [comment]
> This tool should have been light, standalone. But, because of the dependency on libcompose, openshift, kubernetes repos, it has indirectly become a bulkier tool, and this fact shows up in the increased build time of this application, with so much less code on our side.
> 
> All we are doing is taking one file and producing another file from it, kind of a mapping (one-to-one or complicated), but it's at the end, a mapping. 

Yes, and that is what meant by this issue, we need to document how that mapping works, what is mapped to what.
>Because of the huge dependency chain of this library/tool, it will be difficult to promote it as a library for consumption in Openshift.

Right now all our dependencies are from OpenShift, so if we would like to create library from Henge and integrate it back to OpenShift it shouldn't be problem. 
But I don't think that we should put to much focus on creating library right now. Lets use OpenShift as upstream, and build nice command-line tool first.  We can always create library from this later on, if there is demand for it.
> 
> Let me know, what your thoughts are.

**Comment by pradeepto** [comment]
@rtnpro What you are talking about is correct. But that is completely different "task" / "user story" which needs to be tackled separately. This issue is just about documenting the mappings. Consider the output of this task to be an authorative document that will have "how A in Foo tool will map to B in Kubernetes/OpenShift" and alike.

Let's make sure we document the conversion process as honestly as possible. The issues that are caused by our current choice (and not our choice) of libraries etc are beyond the scope of this issue.

**Comment by kadel** [comment]
draft: https://github.com/redhat-developer/henge/wiki/Mappings
